app/models/piece.rb

```diff
@@ -14,7 +14,7 @@ def move_to!(params)
       # Step 2: Check to see if the piece is of the oppposite color.
       if color != enemy_piece[:color]
         # Step 3: Remove piece from the board.
-        enemy_piece.update_attributes(horizontal_position: nil, vertical_position: nil)
+        enemy_piece.destroy
```

Comment:
I don't see a problem with this. The example above doesn't rely on pieces position becoming nil; it is referring to the data which comes from a pawn promotion piece selection. It also functions by updating the piece type, as opposed to removing it from the board and replacing it. I didn't notice any test fails from switching to destroy method either..